From 0981de6ff0a072fd25d919e661ac22890a7a1e34 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Sat, 1 Jan 2022 18:15:03 +0100
Subject: [PATCH] clk: qcom: ipq8074: add missing networking resets

Downstream QCA 5.4 kernel defines networking resets which are not present
in the mainline kernel but are required for the networking drivers.

So, port the downstream resets and avoid using magic values for mask,
construct mask for resets which require multiple bits to be set/cleared.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/clk/qcom/gcc-ipq8074.c               | 14 ++++++++++++++
 include/dt-bindings/clock/qcom,gcc-ipq8074.h | 14 ++++++++++++++
 2 files changed, 28 insertions(+)

--- a/drivers/clk/qcom/gcc-ipq8074.c
+++ b/drivers/clk/qcom/gcc-ipq8074.c
@@ -4791,6 +4791,20 @@ static const struct qcom_reset_map gcc_i
 	[GCC_PCIE1_AHB_ARES] = { 0x76040, 5 },
 	[GCC_PCIE1_AXI_MASTER_STICKY_ARES] = { 0x76040, 6 },
 	[GCC_WCSSAON_RESET] = { 0x59010, 0 },
+	[GCC_PPE_FULL_RESET] = { 0x68014, 0, GENMASK(19, 16) },
+	[GCC_UNIPHY0_SOFT_RESET] = { 0x56004, 0, GENMASK(13, 4) | BIT(1) },
+	[GCC_UNIPHY0_XPCS_RESET] = { 0x56004, 2 },
+	[GCC_UNIPHY1_SOFT_RESET] = { 0x56104, 0, GENMASK(5, 4) | BIT(1) },
+	[GCC_UNIPHY1_XPCS_RESET] = { 0x56104, 2 },
+	[GCC_UNIPHY2_SOFT_RESET] = { 0x56204, 0, GENMASK(5, 4) | BIT(1) },
+	[GCC_UNIPHY2_XPCS_RESET] = { 0x56204, 2 },
+	[GCC_EDMA_HW_RESET] = { 0x68014, 0, GENMASK(21, 20) },
+	[GCC_NSSPORT1_RESET] = { 0x68014, 0, BIT(24) | GENMASK(1, 0) },
+	[GCC_NSSPORT2_RESET] = { 0x68014, 0, BIT(25) | GENMASK(3, 2) },
+	[GCC_NSSPORT3_RESET] = { 0x68014, 0, BIT(26) | GENMASK(5, 4) },
+	[GCC_NSSPORT4_RESET] = { 0x68014, 0, BIT(27) | GENMASK(9, 8) },
+	[GCC_NSSPORT5_RESET] = { 0x68014, 0, BIT(28) | GENMASK(11, 10) },
+	[GCC_NSSPORT6_RESET] = { 0x68014, 0, BIT(29) | GENMASK(13, 12) },
 };
 
 static const struct of_device_id gcc_ipq8074_match_table[] = {
--- a/include/dt-bindings/clock/qcom,gcc-ipq8074.h
+++ b/include/dt-bindings/clock/qcom,gcc-ipq8074.h
@@ -367,5 +367,19 @@
 #define GCC_PCIE1_AXI_MASTER_STICKY_ARES	130
 #define GCC_PCIE0_AXI_SLAVE_STICKY_ARES		131
 #define GCC_WCSSAON_RESET			132
+#define GCC_PPE_FULL_RESET			133
+#define GCC_UNIPHY0_SOFT_RESET			134
+#define GCC_UNIPHY0_XPCS_RESET			135
+#define GCC_UNIPHY1_SOFT_RESET			136
+#define GCC_UNIPHY1_XPCS_RESET			137
+#define GCC_UNIPHY2_SOFT_RESET			138
+#define GCC_UNIPHY2_XPCS_RESET			139
+#define GCC_EDMA_HW_RESET			140
+#define GCC_NSSPORT1_RESET			141
+#define GCC_NSSPORT2_RESET			142
+#define GCC_NSSPORT3_RESET			143
+#define GCC_NSSPORT4_RESET			144
+#define GCC_NSSPORT5_RESET			145
+#define GCC_NSSPORT6_RESET			146
 
 #endif
